controllers/virtualmachine/volumebatch/volumebatch_controller.go

```diff
@@ -44,6 +44,15 @@ import (
 
 const (
 	controllerName = "volumebatch"
+	// In BatchAttachment status, CSI hardcode a volume name entry with :detaching
```

Comment:
We know that something is detaching as soon as we observe something in the attachment status that isn't referenced by the spec anymore. I guess I'm a little surprised that for batch, we depend on them to reconcile things before that is reflected. In the common case, I don't think we'd ever see detaching from them

Reply:
Yeah they added this `detaching` suffix once they detach the disks https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/55fc214db2e7ae6db2fe6aea8d4b5a28ad6fa88b/pkg/syncer/cnsoperator/controller/cnsnodevmbatchattachment/cnsnodevmbatchattachment_helper.go#L244-L249

Yeah unless detaching is stuck with some error, probably batchAttacment update would be too fast that VMOp might not catch it

Reply:
OK. I think it would be incredibly hard to actually happen but maybe we should rename our legacy `:detaching` to like `:legacy-detaching` so we can't have a conflict.